### PR TITLE
adapters/base catalog config updates

### PR DESCRIPTION
**Pull request by colin-rogers-dbt**
- add repro test
- Revert "add repro test"
- update file_format handling and change model config value to catalog_name

resolves #
[docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose) dbt-labs/docs.getdbt.com/#

### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

File: dbt-adapters/src/dbt/adapters/base/impl.py [hunk 1]
```diff
@@ -107,7 +107,7 @@
 if TYPE_CHECKING:
     import agate
 
-
+CATALOG_INTEGRATION_MODEL_CONFIG_NAME = "catalog_name"
```
Comment (mikealfare) [inline dbt-adapters/src/dbt/adapters/base/impl.py]: I would start a `constants.py` file outside of any other module to avoid circular imports. You wouldn't be able to pull this into `BaseRelation` for instance.

File: dbt-bigquery/src/dbt/adapters/bigquery/catalogs/_generic.py [hunk 2]
```diff
@@ -3,7 +3,6 @@
 
 from dbt.adapters.catalogs import (
     CatalogIntegration,
```
Comment (mikealfare) [inline dbt-bigquery/src/dbt/adapters/bigquery/catalogs/_generic.py]: Minor nit, but this could be one line now.

File: dbt-bigquery/tests/functional/adapter/catalog_integration_tests/__init__.py
Comment (mikealfare) [inline dbt-bigquery/tests/functional/adapter/catalog_integration_tests/__init__.py:1]: If you remove the `__init__.py`, the test collector gets confused when naming test classes/methods the same thing. It's an edge case, but keeping it allows us to namespace with directories and file names instead of making long test class names.

File: dbt-bigquery/tests/functional/adapter/catalog_integrations/test_generic_catalog.py [hunk 3]
```diff
@@ -31,7 +31,7 @@ def prefix():
     """
 {{ config(
     materialized='table',
-    catalog='managed_iceberg',
+    catalog_name='managed_iceberg',
```
Comment (mikealfare) [inline dbt-bigquery/tests/functional/adapter/catalog_integrations/test_generic_catalog.py:34]: If we're making this config name configurable with `CATALOG_INTEGRATION_MODEL_CONFIG_NAME`, should we also use it here? Or do we specifically want this to fail in the event that the default name is updated?

File: dbt-bigquery/tests/functional/adapter/catalog_integrations/test_generic_catalog.py [hunk 4]
```diff
@@ -46,7 +46,7 @@ def prefix():
 {{ config(
     materialized="incremental",
     unique_key="id",
-    catalog='managed_iceberg',
+    catalog_name='managed_iceberg',
```
Comment (mikealfare) [inline dbt-bigquery/tests/functional/adapter/catalog_integrations/test_generic_catalog.py:49]: Same question as above.

File: dbt-snowflake/src/dbt/adapters/snowflake/catalogs/_built_in.py [hunk 5]
```diff
@@ -4,19 +4,21 @@
 from dbt.adapters.catalogs import (
     CatalogIntegration,
     CatalogIntegrationConfig,
+    CatalogRelation,
```
Comment (mikealfare) [inline dbt-snowflake/src/dbt/adapters/snowflake/catalogs/_built_in.py]: Since this is a protocol, you shouldn't need to inherit from it. As long as `BuiltInCatalogRelation` implements the protocol, it should pass type checking.

File: dbt-snowflake/src/dbt/adapters/snowflake/catalogs/_info_schema.py [hunk 6]
```diff
@@ -4,18 +4,20 @@
 from dbt.adapters.catalogs import (
     CatalogIntegration,
     CatalogIntegrationConfig,
+    CatalogRelation,
```
Comment (mikealfare) [inline dbt-snowflake/src/dbt/adapters/snowflake/catalogs/_info_schema.py]: Same observation regarding protocols.

File: dbt-adapters/src/dbt/adapters/base/impl.py (hunk 1 above)
Comment (colin-k-rogers) [reply dbt-adapters/src/dbt/adapters/base/impl.py]: I thought of that but it just seemed a little silly to have a constants.py with just one value

File: dbt-bigquery/tests/functional/adapter/catalog_integration_tests/__init__.py
Comment (colin-k-rogers) [reply dbt-bigquery/tests/functional/adapter/catalog_integration_tests/__init__.py:1]: oh cool, I had figured this was just a mistake

File: dbt-bigquery/tests/functional/adapter/catalog_integrations/test_generic_catalog.py (hunk 3 above)
Comment (colin-k-rogers) [reply dbt-bigquery/tests/functional/adapter/catalog_integrations/test_generic_catalog.py:34]: We specifically want it to fail so it's explicit that this is a user facing convention

**Review by VersusFacit — COMMENTED** [review]
LGTM, thank you for good PR writeup
